kernel/sys.c

```diff
@@ -831,6 +836,12 @@ SYSCALL_DEFINE1(setfsgid, gid_t, gid)
  */
 SYSCALL_DEFINE0(getpid)
 {
+#ifdef CONFIG_POPCORN
+	if (distributed_remote_process(current)) {
+		return redirect_getpid(0);
```

Comment:
Again, why we want to forward getpid() back? Is there a scenario that caused errors?